app/controllers/auth_services_controller.rb

```diff
@@ -70,4 +73,25 @@ def omniauth_providers
   def redirect_path
     :services
   end
+
+  def is_viable_redirect_location(url)
```

Comment:
A more Rails way of naming this would be to do something like `viable_redirect_location?(url)`.